### PR TITLE
Implements Yaml based configuration validation

### PR DESCRIPTION
**Pull request by Shivansh-007**
This is an exciting commit! Very generally, the code runs like this:

- It reads through the config, getting only the values that can be configured, it lets the user know when it gets them. If a value is received that can't be configured, it lets the user know and skips it.

- At the moment, only token is configurable, though due to the way it is implemented, it should be extremely easy.

Here is an example config.yaml (they must use the YAML format):
```yml
token: aabbccdd
```

### CODE REVIEW

**Comment by JoeNibe** [comment]
Does the user have to change the YAML file to change the config?

**Comment by Shivansh-007** [comment]
I am changing some things here, to allow optional and required configurations too, so hold on.

And yes, on changing the yaml file the user would also change the configuration.

**Comment by JoeNibe** [comment]
can we add an option in the game menu for this? or it will be difficult to change. maybe keep that option as low priority.

File: app/configuration/config_loader.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            # The path is in a default location, a path was probably not specified,
+            # unless the user pointed to one in the default location
+            not existing_paths
+            and self.paths in Configuration.configuration_paths.values()
+        ):
+            term.red_on_black(("No default configuration file found on your system"))
+            return False
+
+        # Choose the first path which is loaded
+        self.path_loaded = existing_paths[0]
```
Comment (p0lygun) [inline app/configuration/config_loader.py:85]: if there is not file on any of the path save a default yaml file and use that

File: app/configuration/config_loader.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    @staticmethod
+    def exit(term: Terminal) -> None:
+        """Show the invalid configuration exit screen."""
+        print("\n\n")
+        print(
+            term.bold_white_on_red(
+                term.center("Error loading the configuration. Exiting in 10 seconds...")
+            )
+        )
+        time.sleep(10)
```
Comment (p0lygun) [inline app/configuration/config_loader.py:124]: reduce to 5 seconds

File: app/configuration/config_loader.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        # Configuration validator
+        response = fix_config(config, term)
+        if not response:
+            self.exit(term)
+            return False
+
+        print("\n\n")
+        print(
+            term.bold_black_on_green(
+                term.center("Successfully loaded configuration...")
```
Comment (p0lygun) [inline app/configuration/config_loader.py:109]: only show error screen no need to show anything if the Yaml got loaded correctly

File: app/configuration/config_loader.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+        # Choose the first path which is loaded
+        self.path_loaded = existing_paths[0]
+        self._load_yaml(term)  # Set config
+        return self.config
+
+    def config_loader_screen(self, term: Terminal) -> Union[dict, bool]:
+        """Screen showing the erros and configuration loaded by the module."""
+        print(term.home + term.clear + term.move_y(0))
+        print(term.bold_black_on_green(term.center("Loading Configurations")))
```
Comment (p0lygun) [inline app/configuration/config_loader.py:92]: no need to show loading 
only error screen should be shown

File: app/game_manager.py [hunk 5]
```diff
@@ -191,6 +192,14 @@ def start_game(self) -> None:
         if self.show_welcome_screen() == "q":
             print(self.term.clear + self.term.exit_fullscreen)
         else:
+            # Show config loader
+            config = ConfigLoader()
+            response = config.config_loader_screen(self.term)
```
Comment (p0lygun) [inline app/game_manager.py:197]: do this before showing welcome screen 
welcome screen will only be shown if everything if running smoothly

**Review by p0lygun — CHANGES_REQUESTED** [review]
did a quick review 
there is stuff like doc strings that needs to be fixed

File: app/constants.py [hunk 6]
```diff
@@ -34,3 +36,22 @@
     " Settings ": ("Change game settings", "bold_white_on_green"),
     " Exit ": ("Exit the game", "bold_white_on_red"),
 }
+
+
+class Configuration:
+    """Configuration constants for the app."""
+
+    configuration_paths = {
```
[the remark this answers is not in the document]
Comment (Shivansh-007) [reply app/constants.py:49]: Umm, I don't want to add more dependencies to the already heavy environment, so I just did this and it should work on all OS systems, hasn't caused much problem yet.

File: app/configuration/config_loader.py (hunk 1 above)
Comment (Shivansh-007) [reply app/configuration/config_loader.py:85]: The token cannot be default so I don't see why this is needed.

File: app/configuration/config_loader.py (hunk 2 above)
Comment (Shivansh-007) [reply app/configuration/config_loader.py:124]: Alright will do.

File: app/configuration/config_loader.py (hunk 3 above)
Comment (Shivansh-007) [reply app/configuration/config_loader.py:109]: alright 👍🏾

File: app/configuration/config_loader.py (hunk 4 above)
Comment (Shivansh-007) [reply app/configuration/config_loader.py:92]: Umm, I did this to ensure users what they are loading double checking, it is like a debug and it kinda looks plain without knowing what is happening. Like you start putting errors you dunno it happened on loading screen, and you dunno what all loaded successfully and what all haven't been loaded yet.

Comment (p0lygun) [reply app/configuration/config_loader.py:92]: then do this before welcome screen is shown

File: app/game_manager.py (hunk 5 above)
Comment (Shivansh-007) [reply app/game_manager.py:197]: I like the welcome screen as the start of the app, that's how it is done on game apps if I remember correctly.

File: app/configuration/config_loader.py (hunk 4 above)
Comment (Shivansh-007) [reply app/configuration/config_loader.py:92]: See below comment.

File: app/configuration/config_loader.py (hunk 3 above)
Comment (Shivansh-007) [reply app/configuration/config_loader.py]: Fixed in 99541bcb0991efc050bc388dd1d0338cefaf7dd2.

File: app/configuration/config_loader.py (hunk 2 above)
Comment (Shivansh-007) [reply app/configuration/config_loader.py]: Fixed in 93e2d2986b3dad7129d6cab4fea7024c359d0aea.

**Comment by Shivansh-007** [comment]
Closing this as I feel it is redundant and making the user's path complex.

Instead we can just ask the user for token when he plays/creates a game and store it in cache.

**Pull request closed without merging**